### PR TITLE
Adding TGR Tris/Tris CE Numpad

### PR DESCRIPTION
**Pull request by halfenergized**
## Description

Adding TGR Tris/Tris CE Numpad (including [VIA support](https://github.com/the-via/keyboards/pull/144))

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tgr/tris/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (zvecr) [inline keyboards/tgr/tris/keymaps/default/keymap.c:61]:
```suggestion
```

File: keyboards/tgr/tris/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ),
+
+    [2] = LAYOUT_numpad_6x4(
+        RESET,   KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,          KC_TRNS
+    )
```
Comment (drashna) [inline keyboards/tgr/tris/keymaps/default/keymap.c:46]: These are formatted wrong, and will produce the wrong layout when compiled. 

Specifically, this needs to be formatted to match: 
https://github.com/qmk/qmk_firmware/blob/master/layouts/default/numpad_6x4/default_numpad_6x4/keymap.c

```suggestion
    [0] = LAYOUT_numpad_6x4(
        KC_ESC,  KC_TAB,  MO(1),   KC_BSPC,
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
        KC_P0,            KC_PDOT, KC_PENT
    ),

    [1] = LAYOUT_numpad_6x4(
        KC_TRNS,   KC_TRNS, KC_TRNS, MO(2),
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_HOME, KC_UP,   KC_PGUP,
        KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_END,  KC_DOWN, KC_PGDN,
        KC_INS,           KC_DEL,  KC_TRNS
    ),

    [2] = LAYOUT_numpad_6x4(
        RESET,   KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS,          KC_TRNS, KC_TRNS
    )
```

File: keyboards/tgr/tris/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    ),
+
+    [3] = LAYOUT_numpad_6x4(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,          KC_TRNS
+    )
```
Comment (drashna) [inline keyboards/tgr/tris/keymaps/via/keymap.c:55]:
```suggestion
    [0] = LAYOUT_numpad_6x4(
        KC_ESC,  KC_TAB,  MO(1),   KC_BSPC,
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
        KC_P0,            KC_PDOT, KC_PENT
    ),

    [1] = LAYOUT_numpad_6x4(
        KC_TRNS,   KC_TRNS, KC_TRNS, MO(2),
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_HOME, KC_UP,   KC_PGUP,
        KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_END,  KC_DOWN, KC_PGDN,
        KC_INS,           KC_DEL,  KC_TRNS
    ),

    [2] = LAYOUT_numpad_6x4(
        RESET,   KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS,          KC_TRNS, KC_TRNS
    ),

    [3] = LAYOUT_numpad_6x4(
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS,          KC_TRNS, KC_TRNS
    )
```

File: keyboards/tgr/tris/tris.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+#define LAYOUT_numpad_6x4( \
+    k00, k01, k02, k03, \
+    k10, k11, k12, k13, \
+    k20, k21, k22,      \
+    k30, k31, k32, k33, \
+    k40, k41, k42, k43, \
+    k50,      k52       \
+) \
```
Comment (drashna) [inline keyboards/tgr/tris/tris.h:55]:
```suggestion
#define LAYOUT_numpad_6x4( \
    k00, k01, k02, k03, \
    k10, k11, k12, k13, \
    k20, k21, k22,      \
    k30, k31, k32, k33, \
    k40, k41, k42,      \
    k50,      k52, k43  \
) \
```

File: keyboards/tgr/tris/info.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    "maintainer": "halfenergized",
+    "width": 4,
+    "height": 6,
+    "layouts": {
+        "LAYOUT_ortho_6x4": {
+            "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":0, "y":1.25}, {"x":1, "y":1.25}, {"x":2, "y":1.25}, {"x":3, "y":1.25}, {"x":0, "y":2.25}, {"x":1, "y":2.25}, {"x":2, "y":2.25}, {"x":3, "y":2.25}, {"x":0, "y":3.25}, {"x":1, "y":3.25}, {"x":2, "y":3.25}, {"x":3, "y":3.25}, {"x":0, "y":4.25}, {"x":1, "y":4.25}, {"x":2, "y":4.25}, {"x":3, "y":4.25}, {"x":0, "y":5.25}, {"x":1, "y":5.25}, {"x":2, "y":5.25}, {"x":3, "y":5.25}]
+        },
+
+        "LAYOUT_numpad_6x4": {
+            "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":0, "y":1.25}, {"x":1, "y":1.25}, {"x":2, "y":1.25}, {"x":3, "y":1.25}, {"x":0, "y":2.25}, {"x":1, "y":2.25}, {"x":2, "y":2.25}, {"x":3, "y":2.25, "h":2}, {"x":0, "y":3.25}, {"x":1, "y":3.25}, {"x":2, "y":3.25}, {"x":0, "y":4.25}, {"x":1, "y":4.25}, {"x":2, "y":4.25}, {"x":3, "y":4.25, "h":2}, {"x":0, "y":5.25, "w":2}, {"x":2, "y":5.25}]
```
Comment (zvecr) [inline keyboards/tgr/tris/info.json:13]:
```suggestion
            "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":0, "y":1.25}, {"x":1, "y":1.25}, {"x":2, "y":1.25}, {"x":3, "y":1.25}, {"x":0, "y":2.25}, {"x":1, "y":2.25}, {"x":2, "y":2.25}, {"x":0, "y":3.25}, {"x":1, "y":3.25}, {"x":2, "y":3.25}, {"x":3, "y":2.25, "h":2}, {"x":0, "y":4.25}, {"x":1, "y":4.25}, {"x":2, "y":4.25}, {"x":0, "y":5.25, "w":2}, {"x":2, "y":5.25}, {"x":3, "y":4.25, "h":2}]
```

**Comment by halfenergized** [comment]
> have you tested [ae2d683](https://github.com/qmk/qmk_firmware/commit/ae2d6832673985a874f7f43d42bdb56679d851d6) against hardware? moving `k43` in the bottom part of the layout macro is only required if that is its row and col, to which it previously was not. If so, it should be renamed to `k53`.

Yes, I have tested before the commit and it works properly. It is correctly named as `k43` (not `k53`), as a follow up to this changes https://github.com/qmk/qmk_firmware/pull/9139#discussion_r427739926

**Comment by zvecr** [comment]
If it works as it is in the repo right now, then the naming is wrong. `k43` but in location 5,3. Also your via config still has 4,3 which is why i question it being wrong. You can have different physical and electrical layouts and the comment by drashna was about aligning physical, to which in a later commit you then changed electrical.

You either need to:
1. revert ae2d683
2. rename `k43` to `k53` within both parts of the LAYOUT_numpad_6x4 macro (and update your via definition)

**Comment by halfenergized** [comment]
> If it works as it is in the repo right now, then the naming is wrong. `k43` but in location 5,3. Also your via config still has 4,3 which is why i question it being wrong. You can have different physical and electrical layouts and the comment by drashna was about aligning physical, to which in a later commit you then changed electrical.
> 
> You either need to:
> 
> 1. revert [ae2d683](https://github.com/qmk/qmk_firmware/commit/ae2d6832673985a874f7f43d42bdb56679d851d6)
> 2. rename `k43` to `k53` within both parts of the LAYOUT_numpad_6x4 macro (and update your via definition)
> 
> 1 or 2, not what is in this PR right now

Ah, I see. Sorry for the confusion. I have reverted https://github.com/qmk/qmk_firmware/commit/ae2d6832673985a874f7f43d42bdb56679d851d6 and tested the firmware, all normal.

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**